src/tools/ceph_conf.cc

```diff
@@ -198,6 +226,15 @@ int main(int argc, const char **argv)
       string key(val, 0, pos);
       string value(val, pos+1);
       filter_key_value[key] = value;
+    } else if (ceph_argparse_flag(args, i, "-D", "--dump_all", (char*)NULL)) {
+      action = "dumpall";
+    } else if (ceph_argparse_witharg(args, i, &val, "--format", (char*)NULL)) {
+      if (val != "plain" && val != "json" && val != "json-pretty") {
```

Comment:
this repeats the `if-elseif` checks in `dump_all()` , i'd recommend print the error message in it instead.